### PR TITLE
Add ability to track che tasks

### PR DESCRIPTION
**Pull request by RomanNikitenko**
### What does this PR do?
TaskExitedEvent is fired when a che task is completed, so all interested entities have ability to:
- notify user when a che task is completed
- update UI
- clean up resources 

Please see a short demo https://youtu.be/-qafPjoU2dM

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/12838

### CODE REVIEW

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-client.ts [hunk 1]
```diff
@@ -53,4 +69,8 @@ export class CheTaskClientImpl implements CheTaskClient {
     setRunTaskHandler(handler: (id: number, config: TaskConfiguration, ctx?: string) => Promise<void>) {
         this.runTaskHandlers.push(handler);
     }
+
+    setTaskExitedHandler(handler: (id: number) => Promise<void>) {
```
Comment (RomanNikitenko) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-client.ts]: you are right, thank you!

Comment (RomanNikitenko) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-client.ts]: you are right, thank you!

Comment (mmorhun) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-client.ts]: I would say it is more `add` than `set`

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-client.ts [hunk 2]
```diff
@@ -30,14 +32,28 @@ export class CheTaskClientImpl implements CheTaskClient {
...
-            const taskInfo = await taskInfoHandler(id);
-            if (taskInfo) {
-                return taskInfo;
+            try {
+                const taskInfo = await taskInfoHandler(id);
+                if (taskInfo) {
+                    return taskInfo;
+                }
+            } catch (e) {
+                // allow another hanlers to handle request
```
Comment (mmorhun) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-client.ts]: Typo: `hanlers` -> `handlers`

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-task-service.ts [hunk 3]
```diff
@@ -67,6 +71,34 @@ export class CheTaskServiceImpl implements CheTaskService {
...
+
+                    task.fireTaskExited({ taskId: task.id, code: event.code, ctx: runtimeInfo.ctx });
+                    break;
+                }
+            } catch (e) {
+                // allow another hanlers to handle request
+            }
+        }
+
+        if (!exitedTask) {
```
Comment (mmorhun) [inline extensions/eclipse-che-theia-plugin-ext/src/node/che-task-service.ts]: If we move following logic to `break` place, maybe code will be more readable. But up to you.

Comment (RomanNikitenko) [reply extensions/eclipse-che-theia-plugin-ext/src/node/che-task-service.ts]: Done!

**Pull request merged**